### ISSUE TITLE
Retry is not working when it is added in describe block -2nd attempt hangs forever

### ISSUE THREAD

**Issue by cihanaygunes** [cypress-io/cypress#27958]
### Current behavior

Retry option is not working when i add it in describe block. First attempt fails, however during the second attempt, it hangs in there and stays forever doing nothing. Neither it fails the test or passes. Here is what my test looks like.

![image](https://github.com/cypress-io/cypress/assets/74625769/41ce79d6-34d8-41fc-ba3c-5bf502bcee0a)

### Desired behavior

The test should be run and/or fails after some time.

### Test code to reproduce

```js
const story = 'Do something';

describe(
${story},
{
retries: 1,
},
() => {
beforeEach(async () => {
await stageTestData(Scenario.secretWay, house.secretAddress, brand?.name);
});

it('should do some validations', { tags: ['@regression', '@house'] }, () => {
cy.visit('www.something.com');
cy.login();
cy.get('[data-testid=some-secret-id]').click();
cy.get('.serviceCompletion', { timeout: 0 }).get('[data-testid=submit]').should('have.length', 1);
});
});
```

### Cypress Version

13.2.0

### Node version

v20.3.1

### Operating System

macOS Venture - Version 13.6

**Comment by jennifer-shehane** [comment]
@cihanaygunes Is this hanging something that began happening on a more recent version of Cypress? Was this something that was every working?

**Comment by cihanaygunes** [comment]
Hey @jennifer-shehane. No, it is a new initiation. So, it is not something that it was working before.

**Comment by jennifer-shehane** [comment]
@cihanaygunes Did you upgrade from a previous version where this was not happening? We’ll need more info, since we can’t recreate the issue given your example. It’s likely something very specific to your setup since this is not a widespread reported issue.

**Comment by cihanaygunes** [comment]
@jennifer-shehane. Hmm interesting. I remember reading several similar issues although they were in earlier versions. To answer your question; I did not upgrade from a previous version. This was never working because as i mentioned this was not tried with older versions. Unfortunately, I am unable to share more details other than what i provided above (That was the frame of the test). 

FYI - We found a workaround to use cy.request instead of await/async axios call. And moved the call as a command in test's body rather than having it in beforeEach hook. Thank you for your response.
